src/core/net_db.cc

```diff
@@ -154,34 +154,30 @@ void NetDb::Run() {
       }
       // publishes router info to a floodfill at Nth interval
       if (ts - last_publish >=
-	  static_cast<std::uint16_t>(NetDbDuration::PublishRouterInfoInterval)) {
+	  static_cast<std::uint16_t>(NetDbInterval::PublishRouterInfo)) {
         Publish();
         last_publish = ts;
       }
       // builds exploratory tunnels at Nth interval to find more peers
-      if (ts - last_exploratory >=
-          static_cast<std::uint16_t>(NetDbDuration::ExploreInterval)) {
-        auto known_routers = m_RouterInfos.size();
+      if (ts - last_exploratory >= static_cast<std::uint16_t>(NetDbInterval::Exploratory)) {
+        auto known_routers = GetNumRouters();
         std::uint16_t num_routers = 0;
```

Comment:
If I'm not mistaken, this should be initialized with `static_cast<std::uint16_t>(NetDbSize::MinExploratoryTunnels)` and then we can lose the `else if` block entirely (assuming we truly don't care about delayed exploratory time (see comment below)). I don't think we ever should have this as 0 - or else we won't explore.